packages/sdk/src/reads/getUniswapV2PairData.ts

```diff
@@ -0,0 +1,87 @@
+import { type ReadContractParameters, readContractParameters } from "../utils/viem.js";
+import type { Address, PublicClient } from "viem";
+
+const pairAbi = {
+  constant: true,
+  inputs: [
+    { internalType: "address", name: "", type: "address" },
+    { internalType: "address", name: "", type: "address" },
+  ],
+  name: "getPair",
+  outputs: [{ internalType: "address", name: "", type: "address" }],
+  payable: false,
+  stateMutability: "view",
+  type: "function",
+};
+
+const reservesAbi = {
+  constant: true,
+  inputs: [],
+  name: "getReserves",
+  outputs: [
+    {
+      internalType: "uint112",
+      name: "reserve0",
+      type: "uint112",
+    },
+    {
+      internalType: "uint112",
+      name: "reserve1",
+      type: "uint112",
+    },
+    {
+      internalType: "uint32",
+      name: "blockTimestampLast",
+      type: "uint32",
+    },
+  ],
+  payable: false,
+  stateMutability: "view",
+  type: "function",
+};
+
+type Token = {
+  chainId: number;
+  address: Address;
+  decimals: number;
+  symbol?: string;
+  name?: string;
+};
+
+const UniswapV2PairDeployerContract = "0x5C69bEe701ef814a2B6a3EDD4B1652CB9cc5aA6f" as const;
+
+export async function getUniswapV2PairData(
+  client: PublicClient,
+  args: ReadContractParameters<{
+    token0: Token;
```

Comment:
only Token address is used, there is no need for other variables, I would do `token0: Address;`